src/ORM/Table.php

```diff
@@ -788,16 +788,17 @@ public function behaviors(): BehaviorRegistry
      */
     public function getBehavior(string $name): Behavior
     {
-        /** @var \Cake\ORM\Behavior|null $behavior */
-        $behavior = $this->_behaviors->get($name);
-        if ($behavior === null) {
+        if (!$this->_behaviors->has($name)) {
             throw new InvalidArgumentException(sprintf(
                 'The %s behavior is not defined on %s.',
                 $name,
                 static::class
             ));
         }
 
+        /** @var \Cake\ORM\Behavior|null $behavior */
+        $behavior = $this->_behaviors->get($name);
```

Comment:
Why the `|null`? `get()` doesn't return `null` anymore right? Even the assignment is not necessary, can directly return the value:

```
/** @var \Cake\ORM\Behavior */
return $this->_behaviors->get($name);
```

Reply:
True

Reply:
Ah yes since the parent class has `object` as return type. Nevermind then.